.github/workflows/linux.yaml

```diff
@@ -0,0 +1,103 @@
+name: Linux
+
+on: [push, pull_request]
+
+env:
+  REGISTRY_HOST: pcr-internal.puppet.net
+  DOCKER_REGISTRY: pe-and-platform
+  BOLT_SUDO_USER: true
```

Comment:
Like Cas pointed out, we need to make sure we're still pushing a container to the PCR registry on merges to master and new tags. The way this worked in Travis was that we would encrypt the PCR password using this workflow: https://docs.travis-ci.com/user/encryption-keys/#usage, which encrypts a value and makes it available as an environment variable in the CI runner.

It looks like for GH actions we'll need to add the password as a secret to this repo, then add that as the DOCKER_TOKEN environment variable and run the deploy.sh script. https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets#using-encrypted-secrets-in-a-workflow

Since we only want to deploy the container on merge to master or a new tag, I think it might make the most sens to make a separate `deploy.yaml` (or maybe have it in this file? https://github.community/t5/GitHub-Actions/Depend-on-another-workflow/td-p/32317 that might be more trouble than it's worth).

Reply:
Created a `deploy` workflow that triggers on `pull_request:closed` and runs the script if the PR was merged. Working on getting an access token to the docker registry and then I'll get that stored as a secret.

Reply:
I think merge into the master branch might be a better trigger? We don't typically push directly to master but if we did we'd want that to deploy too. Plus we also want new tags to deploy as well.

Reply:
Unfortunately there's no `merge` trigger for actions. The workaround is to use `pull_request:closed` and check that the PR was merged. I can set it up to trigger on `push` to master, though.

Reply:
As for running when a new tag is created... it looks like the `create` trigger covers that, but also gets triggered when new branches of the repo are created. We might be able to get away with this trigger for now, though, since Jenkins pushes to master after tagging, which would trigger the workflow.

Reply:
Ah ok I see - `if: github.event.pull_request.merged`. Yeah, I'm ok with waiting until it's a problem.

Reply:
Doing a little more digging around into other people's workflows, it looks like a push trigger to master should cover merging PRs and direct commits.